### ISSUE TITLE
Abstract function return type is wrongly detected as a constant

### ISSUE THREAD

**Issue by bancer** [cakephp/cakephp-codesniffer#220]
The code:
`abstract protected function _doSomething(string $param) : void;`

The reported error:
```
--------------------------------------------------------------------------------
FOUND 1 ERROR(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 29 | ERROR | Constants must be uppercase; expected VOID but found void
--------------------------------------------------------------------------------
```

PHP 7.1.24
CakePHP 2.10.13
CakePHP Code Sniffer 1.5.6

Similar issue fixed recently - https://github.com/cakephp/cakephp-codesniffer/issues/204.

**Comment by bancer** [comment]
I have no idea how to fix that.

**Comment by dereuromark** [comment]
The sniff just needs to see this as a return type hint, and return early if that is the case.
Most likely this is also fixed in the new branch, maybe it can be ported back if you find a bit of time?

**Comment by dereuromark** [comment]
Is there sth to be done here? People should be using 3.x of code sniffer by now.

**Comment by markstory** [comment]
I'm going to close it. If folks feel really strongly they can comment here and provide a working patch.
